bouncer/src/repo/manipulator/modelconvertor/import/odaHelper/data_processor_nwd.h

```diff
@@ -42,6 +47,23 @@ namespace repo {
 				private:
 					GeometryCollector* collector;
 				};
+
+				class NwdFilePathRemovalVisitor : public boost::static_visitor<> {
```

Comment:
this seems very over engineered for file path removal?

it's unclear to me why we need this. Can't we sanitise the string before we create the variant?

Reply:
Mostly because that's how it was before. Not with the visitor, but cycling over all metadata at the end. Maybe because there is more than one way for a string such as that to enter the metadata. As a minimum, I would need to add  checks for the two labels and sanitation to tryConvert for the property attributes and then to setMetadataValue(std::string, OdString, OdString,...).

With tryConvert moving into the class, this is a possibility, but I see the appeal of a single location for sanitation, though it is not the option with the best performance.

Reply:
considering we know exactly which one we need to santise i feel like it'll make more sense if we do it before we put it into the metadata collection 
![image](https://github.com/user-attachments/assets/417a35cf-1582-4b7d-9826-74a9e2d9f05c)

Could wait for @sebjf's take on it if you like!

Reply:
No it's alright. I don't think this is large enough to block this until he is back.

Reply:
Done in [810b669](https://github.com/3drepo/3drepobouncer/pull/694/commits/810b669513f4efda87b17ba94a6fd6295959cd92) and [4f7ca91](https://github.com/3drepo/3drepobouncer/pull/694/commits/4f7ca91472201b248e0148f1de91cdb661597404)

Reply:
For completeness, I agree that this should be before the variant creation, if only because only strings can represent paths!

FYI that `NwdFilePathRemovalVisitor` is still in the header as of the latest push, I think you intended to remove that @FJThiel!

Finally, I would rename `checkForPathAndSanitise` (e.g. to just `sanitiseString`), as currently it makes it sound like its guessing whether a string is a path and sanitising it if so. We can leave the nature of the sanitisation, including any filtering, as implementation detail of the function (and also add additional sanitisation there in the future if needed).